misc/python/materialize/cloudtest/app/materialize_application.py

```diff
@@ -48,37 +49,23 @@ def __init__(
         self.testdrive = Testdrive(release_mode=release_mode, aws_region=aws_region)
         self.release_mode = release_mode
         self.aws_region = aws_region
+        self.root = ROOT
 
         # Register the VpcEndpoint CRD.
-        self.kubectl(
-            "apply",
-            "-f",
-            os.path.join(
-                os.path.abspath(ROOT),
-                "src/cloud-resources/src/crd/gen/vpcendpoints.json",
-            ),
-        )
+        self.register_vpc_endpoint()
 
-        # Start metrics-server.
-        self.kubectl(
-            "apply",
-            "-f",
-            "https://github.com/kubernetes-sigs/metrics-server/releases/download/metrics-server-helm-chart-3.8.2/components.yaml",
-        )
+        self.start_metrics_server()
 
-        self.kubectl(
-            "patch",
-            "deployment",
-            "metrics-server",
-            "--namespace",
-            "kube-system",
-            "--type",
-            "json",
-            "-p",
-            '[{"op": "add", "path": "/spec/template/spec/containers/0/args/-", "value": "--kubelet-insecure-tls" }]',
-        )
+        self.resources = self.collect_resources(release_mode, log_filter, tag)
+        self.images = self.collect_images()
+
+        super().__init__()
+        super().create()
 
-        self.resources = [
+    def collect_resources(
```

Comment:
The `collect_` prefix here and elsewhere may be superfluous , -- consider just `def resources(` .

`collect_` being a verb sort of implies some active action is being taken. Instead, the resources and images are simply being declared.

Reply:
I cannot remove the prefix completely because then the field will have the same name as the method (`self.images = self.images()`). I changed the prefix to `get_`.